### ISSUE TITLE
Report previous block details on heap underflow

### ISSUE THREAD

**Issue by sigurasg** [google/syzygy#21]
In the last SyzyASAN canary, there are 50 heap underflows reported. From an initial look at this, it's possible that what's really happening is overflow from the preceding block, but we don't seem to report on that block.
It would likely be helpful to report the immediately preceding block at least?

**Comment by sebmarchand** [comment]
Yeah, we could probably use the new Kasko Memory Range functionality to make sure that all the useful data gets included in the minidump (most of the time we already have the content of the previous block, but we don't have its stack traces).

**Comment by chhamilton** [comment]
Yup, that would be easiest. We can also extend the AsanErrorInfo and
crashdata struct if need be to add appropriate new fields (like
previous_block and next_block, to be filled out if needed).
